web/controllers/pdf_conversion_controller.ex

```diff
@@ -1,26 +1,21 @@
 defmodule DailyRoutine.PdfConversionController do
   use DailyRoutine.Web, :controller
 
-
-  def create(conn, %{"routine_id" => routine_id, "routine" => routine_params}, user) do
+  def create(conn, %{"routine_id" => routine_id}, user) do
     routine = Repo.get!(DailyRoutine.Routine, routine_id) |> Repo.preload(:user)
+    task = Task.async(fn -> generate_pdf(routine, user) end)
+    result = Task.await(task)
+    redirect(conn, to: routine_path(conn, :show, routine_id))
 
-    case routine do
-      {:ok, routine} ->
-        #TODO
-        # spawn a task and return :ok
-        task = Task.async(fn -> generate_pdf(routine, user) end)
-        result = Task.await(task)
-
-        redirect(conn, to: routine_path(conn, :index))
-      {_} ->
-        #TODO
-        # spawn a task and return :ok
-        redirect(conn, to: routine_path(conn, :index))
-    end
+    # TODO
+    # Do checking based on if the results exists or not
   end
 
   def generate_pdf(routine, user) do
     2 * 2
   end
+
+  def action(conn, _) do
```

Comment:
It would probably be a good idea to convert this into a `plug` or something similar, since it's repeated in the app multiple times already.